compiler/rustc_hir_analysis/src/check/fn_ctxt/suggestions.rs

```diff
@@ -329,17 +329,58 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         expected_ty_expr: Option<&'tcx hir::Expr<'tcx>>,
     ) -> bool {
         let expr = expr.peel_blocks();
-        if let Some((sp, msg, suggestion, applicability, verbose)) =
+        if let Some((sp, msg, suggestion, applicability, verbose, annotation)) =
             self.check_ref(expr, found, expected)
         {
             if verbose {
                 err.span_suggestion_verbose(sp, &msg, suggestion, applicability);
             } else {
                 err.span_suggestion(sp, &msg, suggestion, applicability);
             }
+            if annotation {
+                let suggest_annotation = match expr.peel_drop_temps().kind {
+                    hir::ExprKind::AddrOf(hir::BorrowKind::Ref, hir::Mutability::Not, _) => "&",
+                    hir::ExprKind::AddrOf(hir::BorrowKind::Ref, hir::Mutability::Mut, _) => "&mut ",
+                    _ => return true,
```

Comment:
This should return false, since we didn't actually suggest anything I think.

Reply:
The `err.span_suggestion` statement above has made some modifications to `err`, I guess that might be the reason why we should return true here as well?